tests/test_readwrite.py

```diff
@@ -368,42 +360,59 @@ def check_compressed(key, value):
     assert_equal(adata, expected)
 
 
-def test_zarr_compression(tmp_path):
-    from numcodecs import Blosc
+@pytest.mark.parametrize("zarr_write_format", [2, 3])
+def test_zarr_compression(tmp_path, zarr_write_format):
+    with ad.settings.override(zarr_write_format=zarr_write_format):
+        pth = str(Path(tmp_path) / "adata.zarr")
+        adata = gen_adata((10, 8))
+        if zarr_write_format == 2 or is_zarr_v2():
+            from numcodecs import Blosc
 
-    pth = str(Path(tmp_path) / "adata.zarr")
-    adata = gen_adata((10, 8))
-    compressor = Blosc(cname="zstd", clevel=3, shuffle=Blosc.BITSHUFFLE)
-    not_compressed = []
+            compressor = Blosc(cname="zstd", clevel=3, shuffle=Blosc.BITSHUFFLE)
+        else:
+            from zarr.codecs import BloscCodec
+
+            compressor = BloscCodec(cname="zstd", clevel=3, shuffle="bitshuffle")
+        not_compressed = []
 
-    ad.io.write_zarr(pth, adata, compressor=compressor)
+        ad.io.write_zarr(pth, adata, compressor=compressor)
 
-    def check_compressed(value, key):
-        if isinstance(value, ZarrArray):
-            if value.shape != ():
-                (read_compressor,) = value.compressors
-                print(read_compressor, key)
+        def check_compressed(value, key):
+            if not isinstance(value, ZarrArray) or value.shape == ():
+                return None
+            (read_compressor,) = value.compressors
+            if zarr_write_format == 2:
                 if read_compressor != compressor:
                     not_compressed.append(key)
+                return None
+            if not isinstance(read_compressor, BloscCodec) or (
+                any(
+                    getattr(read_compressor, attr) != getattr(compressor, attr)
+                    for attr in ["clevel", "cname", "shuffle", "blocksize"]
+                )
+                and compressor.typesize is None
+                and isinstance(read_compressor.typesize, int)
+            ):
```

Comment:
this looks complicated, what does this set of properties mean?

Reply:
Ok I will add a comment but basically the blosc codec is attached to a `typesize` now which corresponds to the `dtype` it uses for decoding.  But of course, this value can change based on what it is decoding.  So ever `read_compressor` that is attached to an `Array` will have a value that varies based on the `array` while the original compressor will not have a proper `typesize`.  I guess to "make things work" I can force the `typesize` to be 1 and this will be cleaner (see https://github.com/zarr-developers/numcodecs/pull/713 for a related bug and perhaps more info).

Reply:
yeah, when the purpose isn’t self-evident, a comment with some link is a good idea!